CleverTap/Runtime/Native/UnityNativeWrapper/Models/UnityNativeAppInboxPersistence.cs

```diff
@@ -0,0 +1,300 @@
+        public static Dictionary<string, string> GetAllMessages()
+        {
+            Dictionary<string, string> dict = new Dictionary<string, string>();
+
+            List<string> ids = GetAllMessageIds();
+
+            for (int i = 0, count = ids.Count; i < count; i++)
+            {
+                string id = ids[i];
+                string msg = GetMessage(id);
+
+                if (!string.IsNullOrEmpty(msg))
+                {
+                    dict[id] = msg;
+                }
+            }
+
+            return dict;
+        }
+
+        /// <summary>
+        /// Marks all stored messages as read.
+        /// </summary>
+        public static void MarkAllAsRead()
+        {
+            List<string> ids = GetAllMessageIds();
+
+            for (int i = 0, count = ids.Count; i < count; i++)
+            {
+                MarkAsRead(ids[i]);
+            }
+        }
+
+        /// <summary>
+        /// Deletes all stored messages and their metadata (IDs and read flags).
+        /// </summary>
+        public static void DeleteAllMessages()
+        {
+            if (_preferenceManager == null)
+            {
+                CleverTapLogger.LogError("UnityNativeInboxPersistence is not initialized. Call Initialize() first.");
+                return;
+            }
+
+            List<string> ids = GetAllMessageIds();
+
+            for (int i = 0, count = ids.Count; i < count; i++)
+            {
+                string id = ids[i];
+                _preferenceManager.DeleteKey(MSG_PREFIX + id);
+                _preferenceManager.DeleteKey(READ_PREFIX + id);
+            }
+
+            _preferenceManager.DeleteKey(INBOX_IDS_KEY);
+        }
+
+        /// <summary>
+        /// Retrieves only the unread messages.
+        /// </summary>
+        /// <returns>A dictionary of unread messages keyed by their IDs.</returns>
+        public static Dictionary<string, string> GetUnreadMessages()
+        {
+            var allMessages = GetAllMessages();
+            var unreadMessages = new Dictionary<string, string>();
+
+            foreach (var kvp in allMessages)
+            {
+                if (!IsRead(kvp.Key))
+                {
+                    unreadMessages[kvp.Key] = kvp.Value;
+                }
+            }
+
+            return unreadMessages;
+        }
+
+        /// <summary>
+        /// Determines if the new message is updated or new by comparing it with the stored version.
+        /// </summary>
+        /// <param name="id">The message ID to check.</param>
+        /// <param name="newMessage">The new message content.</param>
+        /// <returns>True if the message is new or updated; false if it matches the stored version.</returns>
+        public static bool IsUpdatedOrNewMessage(string id, string newMessage)
+        {
+            string existingMessage = GetMessage(id);
+            return existingMessage != newMessage;
```

Comment:
This comparison is likely to fail since it compares the JSON string. The JSON is unordered (non-canonical), so the same JSON can have different string representations.
Why is this check needed? If it must be used, you should compare objects initialized from the JSON that correctly implement Equals.